File: server/alerts.py
import time

class Alerts:
    def __init__(self):
        self._alerts = []

    def push(self, level, message, plugin=None):
        self._alerts.append({
            "level": level,
            "message": message,
            "plugin": plugin,
            "time": time.time()
        })

    def list(self):
        return self._alerts[-50:]

File: plugins/admin/ui/alerts.js
export async function renderAlerts(container) {
  async function load() {
    const alerts = await fetch("/api/admin/alerts").then(r => r.json());

    container.innerHTML = `
      <h2>Alerts</h2>
      ${alerts.map(a => `
        <div>
          <b>${a.plugin || "system"}</b>
          <div>${a.message}</div>
        </div>
      `).join("")}
    `;
  }

  await load();
  setInterval(load, 5000);
}
